src/structify/resources/structure.py

```diff
@@ -276,11 +276,25 @@ def run(  # type: ignore
         token: str = self.run_async(*args, **kwargs)  # type: ignore
         start_time = time.time() if timeout is not None else None
 
+        # Job isn't created immediately
+        for _ in range(100):
```

Comment:
100 is a lot of seconds but okay.

Couldn't you just put this in the same while loop as the rest of the code? That's what you're effectively doing since this will just slip into the next loop.